build/common.go

```diff
@@ -0,0 +1,281 @@
+package build
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/magefile/mage/mg"
+	"github.com/magefile/mage/sh"
+)
+
+var exname string = ""
+
+func getExecutableName(os string, arch string) string {
+	if exname == "" {
+		var err error
+		exname, err = getExecutableFromPluginJSON()
+		if err != nil {
+			exname = "set_exe_name_in_plugin_json" // warning in the final name?
+		}
+	}
+
+	exeName := fmt.Sprintf("%s_%s_%s", exname, os, arch)
+	if "windows" == os {
+		exeName = fmt.Sprintf("%s.exe", exeName)
+	}
+	return exeName
+}
+
+func getExecutableFromPluginJSON() (string, error) {
+	byteValue, err := readFileBytes(path.Join("src", "plugin.json"))
+	if err != nil {
+		return "", err
+	}
+
+	var result map[string]interface{}
+	err = json.Unmarshal([]byte(byteValue), &result)
+	if err != nil {
+		return "", err
+	}
+	return result["executable"].(string), nil
+}
+
+func readFileBytes(file string) ([]byte, error) {
```

Comment:
Why not replace this function with [ioutil.ReadFile](https://golang.org/pkg/io/ioutil/#ReadFile)?